### PR TITLE
tracks types of gardens

### LINKED ISSUE #500: Track types of gardens (permaculture, hydroponic, etc)
We should allow people to select what type each of their gardens is, eg.

```
organic
conventional
container
vertical
greenhouse
rooftop
no-dig
raised bed
wicking bed
permaculture
hydroponic
aquaponic
orchard
food forest
biodynamic
```

They should be able to choose more than one.  Selection should be via a series of checkboxes (I think) on the garden new/edit form.

Note that we probably want these to be actual data in the database so that you can easily see "all food forest gardens" or similar.  See plant_parts for an example of how this might work.  Note that only admins should be able to add/edit/remove garden_types.

### PR DESCRIPTION
**Pull request by brandonbaker40**
Allows users to track which types of garden they have. closes #500 

- Allows users to choose more than one container (type) for their garden
- When creating or updating a garden, the garden form contains checkboxes for containers
- Restricts add/edit/remove actions on containers to admins
- Allows users to assign containers to gardens, through plots
- Allows all users to see which gardens are being planted in that container... seen in the `containers/container-slug` url view
- Adds slug for containers
- Adds basic specs for containers and plots
- Adds CanCan authorization for containers

** 'type' is a reserved keyword, so I thought 'container' would be appropriate. I also use 'plot' as the through model because it described the physical resource without interfering with another model, like 'planting', which wouldn't be appropriate anyway because you can have a garden and a container without having to plant anything in it. At least, that's how I understand it.

/cc @Br3nda @Skud

### CODE REVIEW

**Comment by milesgould** [comment]
You're getting a [test failure](https://travis-ci.org/Growstuff/growstuff/jobs/464710929#L3577), and I can't see the reason from a quick skim:
```
Failures:
  1) home page when anonymous should text "community of food gardeners"
     Failure/Error: it { is_expected.to have_text 'community of food gardeners' }
       expected to find text "community of food gardeners" in "Skip navigation menu Toggle Navigation Search crop database: Crops Browse Crops Seeds Plantings Harvests Containers Community Community Map Browse Members Posts Forums member2044 Profile Gardens Plantings Harvest Seeds Posts Inbox Sign out Welcome to Growstuff (travis), member2044 So far, 9 members have planted 4 crops 1 times in 10 gardens. Plant Harvest Add seeds Post Some of our crops magic bean Planted 1 time Recently Planted magic bean member2053's 1F90PBNH3BP5041XT Recently Harvested magic bean Recently Added: magic bean, magic bean, magic bean, magic bean View all crops » Seeds available to trade magic bean Will trade to: locally magic bean Will trade to: locally View all seeds » Discussion Forums: View all posts »"
     # ./spec/features/home/home_spec.rb:78:in `block (3 levels) in <top (required)>')
```

**Comment by Br3nda** [comment]
@brandonbaker40 I think pr #1511 will merge first, but I think this will have few conflicts.

File: db/migrate/20181129171803_create_containers.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class CreateContainers < ActiveRecord::Migration
```
Comment (Br3nda) [inline db/migrate/20181129171803_create_containers.rb]: This will need updating for Rails 5

File: db/schema.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2018_04_01_220637) do
+ActiveRecord::Schema.define(version: 20181201213214) do
```
Comment (Br3nda) [inline db/schema.rb]: You'll need to rerun db:migrate and commit this file again

**Comment by Br3nda** [comment]
It works - but the naming is a bit confusing.

How about calling them "GardenType" instead of Container? It sounds a bit better for the examples in the original story
```
organic
conventional
container
vertical
greenhouse
rooftop
no-dig
raised bed
wicking bed
permaculture
hydroponic
aquaponic
orchard
food forest
biodynamic
```

**Comment by Br3nda** [comment]
I'm not sure what Plot means. I think it's a word the British use to describe their yard?

**Comment by pozorvlak** [comment]
@Br3nda I assume it's short for "plot of land" - don't you say that in NZ? I'm not aware of it being used to mean "yard" in particular in British English, just any (connected?) area of land. Though a quick Google shows that some people say "[vegetable plot](https://en.oxforddictionaries.com/definition/plot)", which I hadn't encountered before.

[Also, @brandonbaker40 lives in Arizona...]

**Comment by Br3nda** [comment]
Plot makes me think of the narrative in a movie or book. I think there's enough regional English variation that others might find the name of the model confusing.

The PR functions great, the code is predictable (this is very good) ansld other than the model naming it's seemingly good to go now.

**Comment by pozorvlak** [comment]
Yeah, I agree it could probably be clearer, but I'm not sure what to replace it with.

**Comment by Br3nda** [comment]
GardenType model can work. Ot sounds clearer to me.

**Pull request merged**